File: problems/2549-count-distinct-numbers-on-board/solution.java
class Solution {
    public int distinctIntegers(int n) {
        Set<Integer> set = new HashSet<>();
        set.add(n);
        Set<Integer> added = new HashSet<>();
        while(true){
            Set<Integer> next = new HashSet<>();
            for(int x : set){
                for(int i = 1; i <= n; ++i){
                    if(x % i == 1){
                        if(!set.contains(i)){
                            next.add(i);
                        }
                    }
                }
            }
            if(next.isEmpty()){
                break;
            }
            set.addAll(next);
        }
        return set.size();
    }
}

File: problems/2549-count-distinct-numbers-on-board/solution.py
class Solution:
    def distinctIntegers(self, n: int) -> int:
        board = {n}
        added = True
        while added:
            added = False
            new_numbers = set()
            for x in board:
                for i in range(1, n + 1):
                    if x % i == 1 and i not in board:
                        new_numbers.add(i)
                        added = True
            board.update(new_numbers)
        return len(board)